packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -367,47 +313,40 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
     }
   }
 
-  function addAutoImports(path, state) {
-    const imports = getImportNames(path, state);
-    if (isModule(path)) {
-      // import {jsx} from "react";
-      // import {createElement} from "react";
-      const importMap = {};
-
-      imports.forEach(importName => {
-        if (!importMap[importName]) {
-          importMap[importName] = addNamed(
-            path,
-            importName,
-            getSource(state.source, importName),
-            {
-              importedInterop: "uncompiled",
-              ensureLiveReference: true,
-            },
-          ).name;
-        }
-      });
+  function createImportLazily(pass, path, importName, source) {
+    return () => {
+      const actualSource = getSource(source, importName);
+      if (isModule(path)) {
+        let reference = pass.get(
+          `@babel/plugin-react-jsx/imports/${importName}`,
+        );
```

Comment:
I'm just wondering - does keeping this in a state has any benefit here? i mean - wouldn't a closure state do the trick just fine? I don't expect any change here - just wondering if there is something I'm not thinking about

Reply:
The plugin might be cached, so multiple runs of the plugin over the same file could have the same closure. Thus, we would still need to have a `programNode -> imports` `WeakMap` to avoid using imports attatched in a separate Babel run.